### DISCUSSION TITLE
Assessing ENTIRE repository / codebase in Visual Studio

### DISCUSSION THREAD

**Discussion by webdevguyrg** [community/community#135804] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I just read about this update: "GitHub Copilot Enterprise subscribers in Visual Studio can now use GitHub Copilot Chat to get answers enriched with context from their entire repository and Bing search results."

I have a personal subscription to Copilot. Is this available yet for me? If not when will it be available?

**Comment by Nihal-A-Khan** [comment]
Hello @webdevguyrg 

I've found a link that has some documentation on your question and could be a great resource: https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes

There is a section with that site that states "[GitHub Copilot Enterprise](https://docs.github.com/en/enterprise-cloud@latest/copilot/github-copilot-enterprise/overview/about-github-copilot-enterprise) subscribers in Visual Studio can now use GitHub Copilot Chat to get answers enriched with context from their entire repository and Bing search results."

If you have any further questions or if this documentation works for you, please let me know! If it resolves your issue, please mark this response as the answer! Thank you in advance!

**Comment by webdevguyrg** [comment]
You quoted the exact quote I quoted. lol 

Any other thoughts?

**Reply by Nihal-A-Khan** [reply, marked as answer]
As of the update you mentioned, I've read through their update notes and dug through other forums for answers. The conclusion I came upon was that the feature allowing GitHub Copilot Chat to provide answers enriched with context from the entire repository and Bing search results is specifically available for GitHub Copilot Enterprise subscribers.

For personal GitHub Copilot subscriptions, this feature is not yet available per any updates notes. Currently, the advanced context and Bing search capabilities are limited to Enterprise users, reflecting a focus on larger organizational needs and extended integrations.

If you’re using GitHub Copilot personally, you’ll need to wait for future updates to see if and when these features become available for individual subscriptions. GitHub typically announces new features and changes through their official channels, so keeping an eye on their blog or updates page is a good way to stay informed. Hope this helps!

I've also looked into this link: https://github.blog/news-insights/product-news/github-copilot-enterprise-is-now-generally-available/
which furthermore highlights the unique availability pertaining to enterprises.

**Reply by webdevguyrg** [reply]
Thanks!

**Reply by Nihal-A-Khan** [reply]
Glad I could help!

**Reply by ayim** [reply]
I'm just going to double check here, are we already using **#solution** in GitHub Copilot Chat to locally include relevant files from the codebase in your chat context?

**Reply by webdevguyrg** [reply]
#solution doesnt seem to work from my usage

**Reply by ayim** [reply]
@webdevguyrg sorry to hear that, a couple q's because I want to help you out

1) which version of VS22 and which languages are we trying to use #solution for?
2) When you say that #solution doesn't seem to work, what are you expecting to see and what is happening instead?
